### PR TITLE
fix bazel integration test on macOS

### PR DESCRIPTION
**Pull request by MarkusZoppelt**
* set options in GitHub Actions that fix linking errors on macOS CI
* only test `run` in integration tests on macOS. (Skipping `coverage` for now and also  `bundle` since it's not supported yet)

### CODE REVIEW

**Comment by MarkusZoppelt** [comment]
Seems like this issue is only happening on Bazel < 6.0 on macOS.

**Note:** It works fine with [Release 6.0.0-pre.20221020.1](https://github.com/bazelbuild/bazel/releases/tag/6.0.0-pre.20221020.1)

If we cannot find a quick fix/workaround I suggest we close this issue/pr until Bazel 6 is fully released.
@adombeck @fmeum what do you think?

**Comment by MarkusZoppelt** [comment]
Tests should run now, but there seems to be something broken on our macOS runner:
```
ERROR: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cifuzz-bazel-testdata-2811710186/src/parser/BUILD.bazel:14:13: Compiling src/parser/parser_fuzz_test.cpp failed: (Exit 1): cc_wrapper.sh failed: error executing command (from target //src/parser:parser_fuzz_test_raw_) 
  (cd /private/var/tmp/_bazel_runner/b39880c5c307e6627675228a7c35257d/sandbox/darwin-sandbox/1/execroot/__main__ && \
  exec env - \
    LD_LIBRARY_PATH=/Users/runner/work/cifuzz/cifuzz/llvm/lib: \
    PATH=/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cifuzz-1006709927/Library/Caches/bazelisk/downloads/bazelbuild/bazel-6.0.0-pre.20221020.1-darwin-x86_64/bin:/Users/runner/hostedtoolcache/go/1.19.2/x64/bin:/Users/runner/.local/bin:/Users/runner/cifuzz/bin:/Users/runner/work/cifuzz/cifuzz/llvm/bin:/Users/runner/go/bin:/Users/runner/hostedtoolcache/go/1.19.2/x64/bin:/usr/local/lib/ruby/gems/3.0.0/bin:/usr/local/opt/ruby@3.0/bin:/usr/local/opt/pipx_bin:/Users/runner/.cargo/bin:/usr/local/opt/curl/bin:/usr/local/bin:/usr/local/sbin:/Users/runner/bin:/Users/runner/.yarn/bin:/Users/runner/Library/Android/sdk/tools:/Users/runner/Library/Android/sdk/platform-tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/bin:/bin:/usr/sbin:/sbin:/Users/runner/.dotnet/tools:/Users/runner/.ghcup/bin:/Users/runner/hostedtoolcache/stack/2.9.1/x64 \
    *** \
  external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections '-std=c++0x' -MD -MF bazel-out/darwin-opt-ST-00475e028063/bin/src/parser/_objs/parser_fuzz_test_raw_/parser_fuzz_test.d '-frandom-seed=bazel-out/darwin-opt-ST-00475e028063/bin/src/parser/_objs/parser_fuzz_test_raw_/parser_fuzz_test.o' '-DBAZEL_CURRENT_REPOSITORY=""' -iquote . -iquote bazel-out/darwin-opt-ST-00475e028063/bin -iquote external/cifuzz -iquote bazel-out/darwin-opt-ST-00475e028063/bin/external/cifuzz -iquote external/rules_fuzzing -iquote bazel-out/darwin-opt-ST-00475e028063/bin/external/rules_fuzzing -iquote external/bazel_tools -iquote bazel-out/darwin-opt-ST-00475e028063/bin/external/bazel_tools -isystem external/cifuzz -isystem bazel-out/darwin-opt-ST-00475e028063/bin/external/cifuzz -g -Og -UNDEBUG -U_FORTIFY_SOURCE -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION '-fsanitize=fuzzer-no-link' '-fsanitize=address' -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c src/parser/parser_fuzz_test.cpp -o bazel-out/darwin-opt-ST-00475e028063/bin/src/parser/_objs/parser_fuzz_test_raw_/parser_fuzz_test.o)
# Configuration: a7c823158f8a8b8d6accbb315a166b2f06bfd1848cf1d399b22ee21923f75bfa
# Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
src/parser/parser_fuzz_test.cpp:1:10: fatal error: 'assert.h' file not found
#include <assert.h>
         ^~~~~~~~~~
1 error generated.
Target //src/parser:parser_fuzz_test_bin failed to build
```

Any ideas?

**Comment by fmeum** [comment]
Looks like https://apple.stackexchange.com/questions/372032/usr-include-missing-on-macos-catalina-with-xcode-11 is the problem. Could mean that LLVM clang always requires flags to work correctly.

File: integration-tests/bazel/bazel_test.go [hunk 1]
```diff
@@ -25,9 +25,12 @@ func TestIntegration_Bazel_InitCreateRunBundle(t *testing.T) {
 		t.Skip("Using cifuzz with bazel is currently only supported on Unix")
 	}
 
-	// TODO: Fix this test on macOS
 	if runtime.GOOS == "darwin" {
-		t.Skip("Building with bazel is currently broken on our macOS GitHub Action runner")
+		// Use bazel >= 6 on macOS.
+		os.Setenv("USE_BAZEL_VERSION", "6.0.0-pre.20221020.1")
```
Comment (adombeck) [inline integration-tests/bazel/bazel_test.go:30]: Lets explain why we use bazel >=6 here

File: integration-tests/bazel/bazel_test.go [hunk 2]
```diff
@@ -25,9 +25,12 @@ func TestIntegration_Bazel_InitCreateRunBundle(t *testing.T) {
...
+
+		// set CPATH to include /usr/include
+		os.Setenv("CPATH", "`xcrun --show-sdk-path`/usr/include")
```
Comment (adombeck) [inline integration-tests/bazel/bazel_test.go:33]: Same here: I'd like the comment to explain why we set `CPATH`

Comment (MarkusZoppelt) [reply integration-tests/bazel/bazel_test.go]: removed it

**Review by adombeck — APPROVED** [review]

**Pull request merged**